File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/enums/TradeItOrderExpirationType.java
package it.trade.android.sdk.enums;

public enum TradeItOrderExpirationType {
    GOOD_FOR_DAY("day"),
    GOOD_UNTIL_CANCELED("gtc");

    private String expirationValue;

    TradeItOrderExpirationType(String expirationValue) {
        this.expirationValue = expirationValue;
    }

    public String getExpirationValue() {
        return expirationValue;
    }


    public static TradeItOrderExpirationType getExpirationForValue(String actionValue) {
        for (TradeItOrderExpirationType expiration: TradeItOrderExpirationType.values()) {
            if (expiration.expirationValue.equals(actionValue)) {
                return expiration;
            }
        }
        return null;
    }
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/enums/TradeItOrderExpirationType.kt
package it.trade.android.sdk.enums

enum class TradeItOrderExpirationType private constructor(val expirationValue: String) {
    GOOD_FOR_DAY("day"),
    GOOD_UNTIL_CANCELED("gtc");


    companion object {


        fun getExpirationForValue(actionValue: String): TradeItOrderExpirationType? {
            for (expiration in TradeItOrderExpirationType.values()) {
                if (expiration.expirationValue == actionValue) {
                    return expiration
                }
            }
            return null
        }
    }
}
